### PR TITLE
prevent xtermColor8 from being set to black

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

Follow up #11483, addresses #11586

### Approach

force `xtermColor8` to a reasonable color, instead of black we get from https://github.com/mbadolato/iTerm2-Color-Schemes/blob/master/xrdb/Tomorrow%20Night.xrdb

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
🤔 this probably also needs to "fix" xtermColor0 (black) for some themes, e.g. Tomorrow Night Bright: 

image

as opposed to e.g. Ambiance where black is set to a different color to make it visible: 

image

File: src/cpp/session/resources/themes/compile-themes.R [hunk 1]
```diff
@@ -695,6 +695,11 @@
        colors <- xterm16ColorMap[order(as.numeric(names(xterm16ColorMap)))]
      }
 
+     # fix themes that incorrectly set silver (xtermColor8) to black
+     if (identical(colors[["8"]], "#000000")) {
```
Comment (romainfrancois) [inline src/cpp/session/resources/themes/compile-themes.R:699]: Maybe instead of this, we could edit the `.xrdb` files.

**Comment by romainfrancois** [comment]
I'm wondering if we should fix those in the 16 that are difficult to see. 

... or if we should decouple the editor theme and a console output theme to govern these 16 colors, based on the many choices available here: https://github.com/mbadolato/iTerm2-Color-Schemes/tree/master/xrdb

cc @DavisVaughan @kevinushey 

For example the "shades of purple" https://github.com/mbadolato/iTerm2-Color-Schemes/blob/master/xrdb/shades-of-purple.xrdb

```
#define Ansi_0_Color #000000
#define Ansi_1_Color #d90429
#define Ansi_10_Color #43d426
#define Ansi_11_Color #f1d000
#define Ansi_12_Color #6871ff
#define Ansi_13_Color #ff77ff
#define Ansi_14_Color #79e8fb
#define Ansi_15_Color #ffffff
#define Ansi_2_Color #3ad900
#define Ansi_3_Color #ffe700
#define Ansi_4_Color #6943ff
#define Ansi_5_Color #ff2c70
#define Ansi_6_Color #00c5c7
#define Ansi_7_Color #c7c7c7
#define Ansi_8_Color #686868
#define Ansi_9_Color #f92a1c
#define Background_Color #1e1d40
#define Badge_Color #ff2600
#define Bold_Color #f9fdff
#define Cursor_Color #fad000
#define Cursor_Guide_Color #b3ecff
#define Cursor_Text_Color #fefff4
#define Foreground_Color #ffffff
#define Link_Color #005cbb
#define Selected_Text_Color #c2c2c2
#define Selection_Color #b362ff
```

Another thing that this may fix is that is has `Link_Color` which we could use for our hyperlinks instead of just not coloring them as we currently do. 

I believe decoupling editor and console theme would give more flexibility to @jimhester's  #4076 approach that we rebooted in #11483, the `Options > Appearance` dialog would then naturally get a "Console theme" below the "Editor theme": 

image

**Comment by romainfrancois** [comment]
Maybe we can hot fix these few issues, i.e. black in Tomorrow Night Bright etc ... here, and perhaps reconsider later if we want to decouple or not.

**Review by kevinushey — APPROVED** [review]
LGTM; is there anything else we want to tweak for this PR?

**Comment by romainfrancois** [comment]
Thanks, i'll finish this (the hotfixes) in the morning and think some more about console themes later. 


One reason we might *not* want console themes such as shades of purple etc ... is that we typically get the colors by e.g crayon::red() or cli::col_red() etc ... so it is expected that those are in the realm of red, rather than color1, color2 ...

**Comment by romainfrancois** [comment]
After some fine tuning, here we are. Most of the changes consisted of using a default color (from the default dark 16 or the default bright 16), but for vibrant, I picked a bright blue manually. 

cobalt: 

image

Dracula: 

image

material: 

image

monokai: 

image

pastel on dark: 

image

solarized dark: 

image

solarized light: 

image

tomorrow: 

image

tomorrow night: 

image

tomorrow night blue:

image

tomorrow night bright: 

image

tomorrow night 80s:

image

twilight:

image

vibrant ink:

image

**Comment by avraam-inside** [comment]
Hi!
I use the Cobalt theme, green used to be green, and now for some reason green is blue. What for???

Return it as it was for cobalt, please

**Pull request merged**